### ISSUE TITLE
Runtime Configuration: Schema Validation Enabled error when message is received  

### ISSUE THREAD

**Issue by supasteri** [wso2/product-apim#12163]
### Description:
I have enable schema validation for a service, as follows: 

![image](https://user-images.githubusercontent.com/13227259/146983026-972e5d05-3921-473f-b205-d12f558892a8.png)

When I try to call the service using the built in test harness, the logs generate the following error.
`
[2021-12-21 20:43:48,541] ERROR - NativeWorkerPool Uncaught exception
java.lang.LinkageError: loader constraint violation: when resolving method "com.github.fge.jsonschema.core.processing.CachingProcessor.<init>(Lcom/github/fge/jsonschema/core/processing/Processor;Lcom/google/common/base/Equivalence;I)V" the class loader (instance of org/eclipse/osgi/internal/loader/EquinoxClassLoader) of the current class, com/github/fge/jsonschema/processors/validation/ValidationChain, and the class loader (instance of org/eclipse/osgi/internal/loader/EquinoxClassLoader) for the method's defining class, com/github/fge/jsonschema/core/processing/CachingProcessor, have different Class objects for the type com/google/common/base/Equivalence used in the signature

	at com.github.fge.jsonschema.processors.validation.ValidationChain.<init>(ValidationChain.java:74) ~[json-schema-validator-2.2.11.jar:?]
	at com.github.fge.jsonschema.main.JsonSchemaFactory.buildProcessor(JsonSchemaFactory.java:259) ~[json-schema-validator-2.2.11.jar:?]
	at com.github.fge.jsonschema.main.JsonSchemaFactory.<init>(JsonSchemaFactory.java:140) ~[json-schema-validator-2.2.11.jar:?]
`
This worked perfectly in WSO2 API Manager 4.0.0

### Steps to reproduce:
Enable schema validation for any service and try to call it. 

### Affected Product Version:

WSO2 API Manager 4.1.0-m2. 
 
### Environment details (with versions):
- OS: Ubuntu 20.04
- Client:
- Env (Docker/K8s): Native

---

**Comment by HiranyaKavishani** [comment]
Hi @supasteri, Could you please attached the swagger file here you used? This cannot be reproduced in default api swagger flow.

**Comment by rmsamitha** [comment]
@supasteri Kind reminder

**Comment by Akila94** [comment]
I also get this error on APIM 4.0.0 as well. I took the updated pack today and encountered the following error when testing with schema validation enabled.

```
[2023-06-15 11:23:19,059] ERROR - NativeWorkerPool Uncaught exception
java.lang.LinkageError: loader constraint violation: when resolving method 'void com.github.fge.jsonschema.core.processing.CachingProcessor.<init>(com.github.fge.jsonschema.core.processing.Processor, com.google.common.base.Equivalence, int)' the class loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @687ebbbf of the current class, com/github/fge/jsonschema/processors/validation/ValidationChain, and the class loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @3ec1e04d for the method's defining class, com/github/fge/jsonschema/core/processing/CachingProcessor, have different Class objects for the type com/google/common/base/Equivalence used in the signature (com.github.fge.jsonschema.processors.validation.ValidationChain is in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @687ebbbf, parent loader 'app'; com.github.fge.jsonschema.core.processing.CachingProcessor is in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @3ec1e04d, parent loader 'app')
	at com.github.fge.jsonschema.processors.validation.ValidationChain.<init>(ValidationChain.java:74) ~[json-schema-validator-2.2.11.jar:?]
	at com.github.fge.jsonschema.main.JsonSchemaFactory.buildProcessor(JsonSchemaFactory.java:259) ~[json-schema-validator-2.2.11.jar:?]
	at com.github.fge.jsonschema.main.JsonSchemaFactory.<init>(JsonSchemaFactory.java:140) ~[json-schema-validator-2.2.11.jar:?]
	at com.github.fge.jsonschema.main.JsonSchemaFactoryBuilder.freeze(JsonSchemaFactoryBuilder.java:139) ~[json-schema-validator-2.2.11.jar:?]
	at com.atlassian.oai.validator.schema.SwaggerV20Library.schemaFactory(SwaggerV20Library.java:67) ~[swagger-request-validator-core_2.12.1.wso2v2.jar:?]
	at com.atlassian.oai.validator.schema.SchemaValidator.<init>(SchemaValidator.java:94) ~[swagger-request-validator-core_2.12.1.wso2v2.jar:?]
	at com.atlassian.oai.validator.OpenApiInteractionValidator.<init>(OpenApiInteractionValidator.java:149) ~[swagger-request-validator-core_2.12.1.wso2v2.jar:?]
	at com.atlassian.oai.validator.OpenApiInteractionValidator.<init>(OpenApiInteractionValidator.java:46) ~[swagger-request-validator-core_2.12.1.wso2v2.jar:?]
	at com.atlassian.oai.validator.OpenApiInteractionValidator$Builder.build(OpenApiInteractionValidator.java:543) ~[swagger-request-validator-core_2.12.1.wso2v2.jar:?]
	at org.wso2.carbon.apimgt.gateway.handlers.security.SchemaValidator.getOpenAPIValidator_aroundBody0(SchemaValidator.java:56) ~[org.wso2.carbon.apimgt.gateway_9.0.174.332.jar:?]
	at org.wso2.carbon.apimgt.gateway.handlers.security.SchemaValidator.getOpenAPIValidator(SchemaValidator.java:47) ~[org.wso2.carbon.apimgt.gateway_9.0.174.332.jar:?]
	at org.wso2.carbon.apimgt.gateway.handlers.security.SchemaValidator.handleRequest_aroundBody2(SchemaValidator.java:67) ~[org.wso2.carbon.apimgt.gateway_9.0.174.332.jar:?]
	at org.wso2.carbon.apimgt.gateway.handlers.security.SchemaValidator.handleRequest(SchemaValidator.java:60) ~[org.wso2.carbon.apimgt.gateway_9.0.174.332.jar:?]
	at org.apache.synapse.api.API.process(API.java:389) ~[synapse-core_2.1.7.wso2v227_55.jar:2.1.7-wso2v227.55]
	at org.apache.synapse.api.AbstractApiHandler.apiProcessNonDefaultStrategy(AbstractApiHandler.java:108) ~[synapse-core_2.1.7.wso2v227_55.jar:2.1.7-wso2v227.55]
	at org.apache.synapse.api.AbstractApiHandler.identifyAPI(AbstractApiHandler.java:128) ~[synapse-core_2.1.7.wso2v227_55.jar:2.1.7-wso2v227.55]
	at org.apache.synapse.api.AbstractApiHandler.dispatchToAPI(AbstractApiHandler.java:60) ~[synapse-core_2.1.7.wso2v227_55.jar:2.1.7-wso2v227.55]
	at org.apache.synapse.api.rest.RestRequestHandler.dispatchToAPI(RestRequestHandler.java:84) ~[synapse-core_2.1.7.wso2v227_55.jar:2.1.7-wso2v227.55]
	at org.apache.synapse.api.rest.RestRequestHandler.process(RestRequestHandler.java:70) ~[synapse-core_2.1.7.wso2v227_55.jar:2.1.7-wso2v227.55]
	at org.apache.synapse.rest.RESTRequestHandler.process(RESTRequestHandler.java:54) ~[synapse-core_2.1.7.wso2v227_55.jar:2.1.7-wso2v227.55]
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.injectMessage(Axis2SynapseEnvironment.java:344) ~[synapse-core_2.1.7.wso2v227_55.jar:2.1.7-wso2v227.55]
	at org.apache.synapse.core.axis2.SynapseMessageReceiver.receive(SynapseMessageReceiver.java:101) ~[synapse-core_2.1.7.wso2v227_55.jar:2.1.7-wso2v227.55]
	at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:180) ~[axis2_1.6.1.wso2v63_6.jar:?]
	at org.apache.synapse.transport.passthru.ServerWorker.processNonEntityEnclosingRESTHandler(ServerWorker.java:393) ~[synapse-nhttp-transport_2.1.7.wso2v227_56.jar:?]
	at org.apache.synapse.transport.passthru.ServerWorker.processEntityEnclosingRequest(ServerWorker.java:452) ~[synapse-nhttp-transport_2.1.7.wso2v227_56.jar:?]
	at org.apache.synapse.transport.passthru.ServerWorker.run(ServerWorker.java:200) ~[synapse-nhttp-transport_2.1.7.wso2v227_56.jar:?]
	at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172) [axis2_1.6.1.wso2v63_6.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
```

**Comment by chamilaadhi** [comment]
This is fixed in the latest 4.1.0 update level.
